rust/nvtcache/tests/nvtcache_tests.rs

```diff
@@ -0,0 +1,131 @@
+        let n = nvtcache::NvtCache::init(&redis_socket, &plugin_path);
+        match n {
+            Ok(nc) => nvtcache = nc,
+            Err(e) => {
+                println!("{}", e);
+                panic!("Error")
+            }
+        }
+
+        if nvtcache.is_init() {
+            println!("Is initialized");
+        }
+
+        match nvtcache.cache.get_namespace() {
+            Ok(ok) => println!("The namespace: {}", ok),
+            Err(e) => println!("Error:{}", e),
+        }
+
+        let mut key = "key int value";
+        let val: u64 = 42;
+        match nvtcache.cache.redis_set_key(key, val) {
+            Ok(_) => println!("Key {} set with {}", key, val),
+            Err(e) => println!("Error:{}", e),
+        }
+        let res = nvtcache.cache.redis_get_key(key);
+        match res {
+            Ok(k) => println!("{}", k),
+            Err(_) => println!("Error"),
+        }
+
+        key = "key string value";
+        let val = "Some string";
+        match nvtcache.cache.redis_set_key(key, val) {
+            Ok(_) => println!("Key {} set with {}", key, val),
+            Err(e) => println!("Error:{}", e),
+        }
+
+        let res = nvtcache.cache.redis_get_key(key);
+        match res {
+            Ok(k) => println!("{}", k),
+            Err(_) => println!("Error"),
+        }
+
+        let _ = nvtcache.set_version("202212101125");
+        let updated = nvtcache.check_feed("202212101125");
+        match updated {
+            Ok(ret) => {
+                assert_eq!(ret, true);
+                println!("Feed up-to-date");
+            }
+            Err(_) => println!("Error"),
+        }
+
+        let mut fake_nvt: Nvt;
+        let res = Nvt::new();
+        match res {
+            Ok(ok) => fake_nvt = ok,
+            Err(_) => panic!("No Nvt"),
+        }
+
+        let oid = "1234".to_owned();
+        match fake_nvt.set_oid(oid) {
+            Ok(_) => (),
+            Err(_) => println!("Error"),
+        }
+        fake_nvt.set_name("Custom Script for the vulnerability 1".to_owned());
+
+        let pref = NvtPref::new(
+            0,
+            "entry".to_string(),
+            "Timeout".to_string(),
+            "320".to_string(),
+        )?;
+        fake_nvt.add_pref(pref);
+
+        //Add first tag
+        fake_nvt.add_tag("Tag Name".to_string(), "Tag Value".to_string());
+        let tag = fake_nvt.get_tag();
+        let expected = vec![("Tag Name".to_string(), "Tag Value".to_string())];
+        assert_eq!(tag, &expected);
+
+        //Add second tag cvss_base, which is ignored
+        fake_nvt.add_tag("cvss_base".to_string(), "Tag Value1".to_string());
+        let tag = fake_nvt.get_tag();
+        let expected = vec![("Tag Name".to_string(), "Tag Value".to_string())];
+        assert_eq!(tag, &expected);
+
+        let filename = "custom.nasl".to_owned();
+        match nvtcache.add_nvt(fake_nvt, filename) {
```

Comment:
I think it would be more readable when you use ? operation and let it fail instead of println